src/colmap/geometry/rigid3.h

```diff
@@ -98,20 +98,45 @@ inline Rigid3d Inverse(const Rigid3d& b_from_a) {
   return a_from_b;
 }
 
-// Update covariance (6x6) for rigid3d.inverse()
+// Update covariance (6 x 6) for rigid3d.inverse()
 //
 // [Reference] Joan Solà, Jeremie Deray, Dinesh Atchuthan, A micro Lie theory
 // for state estimation in robotics, 2018.
 // In COLMAP we follow the left convention (rather than the right convention as
 // in the reference paper and GTSAM). With the left convention the Jacobian of
-// the inverse is -Ad(X^-1). This can be easily derived combining Eqs. (62) and
-// (57) in the paper.
+// the inverse is -Ad(X^{-1}). This can be easily derived combining Eqs. (62)
+// and (57) in the paper.
 inline Eigen::Matrix6d GetCovarianceForRigid3dInverse(
     const Rigid3d& rigid3, const Eigen::Matrix6d& covar) {
   const Eigen::Matrix6d adjoint_inv = rigid3.AdjointInverse();
   return adjoint_inv * covar * adjoint_inv.transpose();
 }
 
+// Given a (12 x 12) covariance on top of two rigid3d objects, a_from_b,
+// b_from_c, this function calculates the (6 x 6) covariance of the composed
+// transformation a_from_c (a_T_b * b_T_c). b_T_c does not contribute to the
+// covariance propagation and is thus not required.
+inline Eigen::Matrix6d GetCovarianceForComposedRigid3d(
+    const Rigid3d& a_from_b, const Eigen::Matrix<double, 12, 12>& covar) {
```

Comment:
nit: for consistency of notation with what we did in `https://github.com/colmap/colmap/blob/main/src/pycolmap/estimators/cost_functions.cc` ?

```suggestion
    const Rigid3d& a_from_b, const Eigen::Matrix<double, 12, 12>& a_cov_from_b) {
```

Reply:
Same for other "covar" variable names.

Reply:
But this is the joint covariance (12 x 12) on a_T_b and b_T_c. Do you think this is not explained in the comments? : )

Reply:
On the other hand, I have fixed the convention in the tests.

Reply:
Right, didn't look carefully enough. Maybe we can be more explicit in the comments.
```
[a_cov_from_b, b_cov_from_a]
[a_cov_from_c, b_cov_from_c]
```
Hope I got it right (on my phone).

Reply:
The understanding is correct, but in your proposal its a 6 x 12 matrix? You have overlooked the correlation part, which is a bit hard to describe in the comments.

Reply:
I made a minor improvement on the comments. Hope it looks better.

Reply:
Wait. But this is not correct then? b_cov_from_a does not equal to the correlation of a_from_b and b_from_c.

Reply:
Still on the phone. Given how tricky this is, we should be very explicit about which covariance is expected here. We shouldn't expect people to go to the paper to figure this out.

Reply:
I am not saying that what I wrote above is correct. It's probably wrong but I want us to be explicit.

Reply:
Its not in the paper. In the relative pose for example, the covariance is defined on (a_from_world, and b_from_world), and the off-diagonal would be just the result of the function we already have here: https://github.com/colmap/colmap/blob/main/src/colmap/estimators/covariance.h#L66. 

However, I see the issue now. The naming of the function is wrong, as it is not Cam2FromCam1Cov. Instead, it is the correlation of cam1_from_world and cam2_from_world, which is a different thing.

Reply:
> Its not in the paper. In the relative pose for example, the covariance is defined on (a_from_world, and b_from_world), and the off-diagonal would be just the result of the function we already have here: https://github.com/colmap/colmap/blob/main/src/colmap/estimators/covariance.h#L66.
> 
> However, I see the issue now. The naming of the function is wrong, as it is not Cam2FromCam1Cov. Instead, it is the correlation of cam1_from_world and cam2_from_world, which is a different thing.

We should fix that naming in a separate PR, once we have a good idea how to name it.

Reply:
PR created here: https://github.com/colmap/colmap/pull/3170